BridgeCareApp/BridgeCare/Services/SummaryReport/SummaryReportBridgeData.cs

```diff
@@ -0,0 +1,361 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using BridgeCare.Interfaces;
+using BridgeCare.Models;
+using OfficeOpenXml;
+using OfficeOpenXml.Style;
+
+namespace BridgeCare.Services
+{
+    public class SummaryReportBridgeData
+    {
+        private readonly IBridgeData bridgeData;
+
+        public SummaryReportBridgeData(IBridgeData bridgeData)
+        {
+            this.bridgeData = bridgeData;
+        }
+
+        /// <summary>
+        /// Fill Summary Report Bridge Data tab.
+        /// </summary>
+        /// <param name="worksheet"></param>
+        /// <param name="simulationModel"></param>
+        /// <param name="simulationYears"></param>
+        /// <param name="dbContext"></param>
+        public void Fill(ExcelWorksheet worksheet, SimulationModel simulationModel, List<int> simulationYears, BridgeCareContext dbContext)
+        {
+            var BRKeys = new List<int>();
+
+            var sections = bridgeData.GetSectionData(simulationModel, dbContext);
+            var simulationDataTable = bridgeData.GetSimulationData(simulationModel, dbContext, simulationYears);
+            var projectCostModels = bridgeData.GetReportData(simulationModel, dbContext, simulationYears);
+            var sectionIdsFromSimulationTable = from dt in simulationDataTable.AsEnumerable()
+                                                select dt.Field<int>("SECTIONID");
+            var sectionsForSummaryReport = sections.Where(sm => sectionIdsFromSimulationTable.Contains(sm.SECTIONID)).ToList();
+            BRKeys = sectionsForSummaryReport.Select(sm => Convert.ToInt32(sm.FACILITY)).ToList();
+            var bridgeDataModels = bridgeData.GetBridgeData(BRKeys, dbContext);
+                        
+            var simulationDataModels = GetSimulationDataModels(simulationDataTable, simulationYears, projectCostModels);
+
+            // Fiil up the excel.
+            var headers = GetHeaders();
+            var currentCell = AddHeadersCells(worksheet, headers, simulationYears);
+
+            // Add row next to headers for fitlers and year no.s for dynamic data. Cover from top, left to right bottom whole set of data
+            using (ExcelRange autoFilterCells = worksheet.Cells[3, 1, currentCell.Row, currentCell.Coln])
+            {
+                autoFilterCells.AutoFilter = true;
+            }
+
+            AddBridgeDataModelsCells(worksheet, bridgeDataModels, currentCell);
+            AddDynamicDataCells(worksheet, sectionsForSummaryReport, simulationDataModels, bridgeDataModels, currentCell);
+            // TODO uncomment later: Postman getting hang if this is kept 
```

Comment:
// TODO The line below currently hangs Postman in testing. It will be required for final production.

Reply:
Updated.